test/unit/RelayServer.test.ts

```diff
@@ -1,688 +1,535 @@
+      );
+    });
+
+    it('Should not charge fees when is sponsored', async function () {
+      mockServer.expects('isSponsorshipAllowed').returns(true);
+
+      const maxPossibleGas = await relayServer.getMaxPossibleGas({
+        relayRequest: {
+          request: {
+            tokenContract: constants.AddressZero,
+          },
+          relayData: {
+            gasPrice: GAS_PRICE,
+          },
+        },
+      } as EnvelopingTxRequest);
+
+      expect(maxPossibleGas.toString()).to.be.equal(
+        fakeEstimationBeforeFees.toString()
+      );
     });
 
-    describe('findMaxPossibleGasWithViewCall', function () {
-        const maxPossibleGas = '115000';
-        let server: RelayServer;
+    describe('When is not sponsored', function () {
+      beforeEach(function () {
+        mockServer.expects('isSponsorshipAllowed').returns(false);
+      });
+
+      it('Should charge fees based on gas when transferFeePercentage = 0', async function () {
+        const fakeServerConfigParams = {
+          app: {
+            gasFeePercentage: FAKE_GAS_FEE_PERCENTAGE,
+            transferFeePercentage: 0,
+          },
+        };
+
+        sinon.stub(relayServer, 'config').value(fakeServerConfigParams);
 
-        const req: RelayTransactionRequest = {
+        const maxPossibleGas = await relayServer.getMaxPossibleGas({
+          relayRequest: {
+            request: {
+              tokenContract: constants.AddressZero,
+              tokenAmount: TOKEN_AMOUNT_IN_REQUEST,
+            },
+            relayData: {
+              gasPrice: GAS_PRICE,
+            },
+          },
+        } as EnvelopingTxRequest);
+
+        expect(maxPossibleGas.toString()).to.be.eq(
+          (
+            fakeEstimationBeforeFees +
+            fakeEstimationBeforeFees * FAKE_GAS_FEE_PERCENTAGE
+          ).toString()
+        );
+      });
+
+      it('Should charge fees based on gas when transferFeePercentage is not defined', async function () {
+        const fakeServerConfigParams = {
+          app: {
+            gasFeePercentage: FAKE_GAS_FEE_PERCENTAGE,
+          },
+        };
+
+        sinon.stub(relayServer, 'config').value(fakeServerConfigParams);
+
+        const maxPossibleGas = await relayServer.getMaxPossibleGas({
+          relayRequest: {
+            request: {
+              tokenContract: constants.AddressZero,
+              tokenAmount: TOKEN_AMOUNT_IN_REQUEST,
+            },
+            relayData: {
+              gasPrice: GAS_PRICE,
+            },
+          },
+        } as EnvelopingTxRequest);
+
+        expect(maxPossibleGas.toString()).to.be.eq(
+          (
+            fakeEstimationBeforeFees +
+            fakeEstimationBeforeFees * FAKE_GAS_FEE_PERCENTAGE
+          ).toString()
+        );
+      });
+
+      it('Should charge fees based on transfer value, when transferFeePercentage > 0 and a transfer() is being relayed', async function () {
```

Comment:
I'd include a `describe` here to join all the tests related to `Should charge fees based on transfer value, when transferFeePercentage > 0`

Reply:
Done.